### PR TITLE
Clean up some helpers

### PR DESCRIPTION
**Pull request by paulmelnikow**
Address comments from #1109.

### CODE REVIEW

File: lib/github-auth.js [hunk 1]
```diff
@@ -120,7 +120,7 @@ function sendTokenToAllServers(token) {
         strictSSL: false,
       };
       request(options, function(err, res, body) {
-        if (err != null) { return reject('Posting the GitHub user token failed: ' + err.stack); }
+        if (err != null) { return reject(err); }
```
Comment (Daniel15) [inline lib/github-auth.js]: Could do `return reject(new Error('Posting the GitHub user token failed: ' + err.message))` or something similar. Up to you.

File: lib/nexus-version.js [hunk 2]
```diff
@@ -1,12 +1,8 @@
...
 
 function isSnapshotVersion(version) {
-  const pattern = /(\d+\.)*\d\-SNAPSHOT/;
-  if (version) {
-    return version.match(pattern);
-  } else {
-    return false;
-  }
+  const pattern = /(\d+\.)*\d-SNAPSHOT/;
+  return version && version.match(pattern);
```
Comment (Daniel15) [inline lib/nexus-version.js]: Nice.

File: lib/regular-update.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ const request = require('request');
 
 // Map from URL to { timestamp: last fetch time, interval: in milliseconds,
 // data: data }.
-let regularUpdateCache = Object.create(null);
+let regularUpdateCache = new Map();
```
Comment (Daniel15) [inline lib/regular-update.js]: All `regularUpdateCache[url]` need to be changed to either `regularUpdateCache.get(url)` or `regularUpdateCache.set(url, value)` in order to use `Map`

File: lib/github-auth.js (hunk 1 above)
Comment (paulmelnikow) [reply lib/github-auth.js]: Yea… I thought about that. This is called in a loop, and the calling code already has something really similar to that. It feels redundant to prefix it twice.

**Comment by paulmelnikow** [comment]
Haha, thanks for the education on `Map`! 😉

File: lib/regular-update.js (hunk 3 above)
Comment (paulmelnikow) [reply lib/regular-update.js]: I'm going to reset these changes.

**Pull request merged**